### PR TITLE
Feature - Cached Tiles

### PR DESCRIPTION
**Pull request by szmoore**
This addresses #46 by caching to a temporary directory.

### CODE REVIEW

**Comment by szmoore** [comment]
I believe this differs from #57 because it caches to files on disk rather than memory. For my usage the cache on disk is more useful, but I haven't explored if `CacheControl` can do this already.

File: contextily/tile.py [hunk 1]
```diff
@@ -162,10 +171,30 @@ def bounds2img(w, s, e, n, zoom='auto',
...
+            image = np.asarray(image)
+            return image
+        except (IOError, OSError, FileNotFoundError) as e:
+            pass
+
     request = _retryer(tile_url, wait, max_retries)
     with io.BytesIO(request.content) as image_stream:
         image = Image.open(image_stream).convert('RGB')
+        if cached_path:
+            if not os.path.exists(os.path.dirname(cached_path)):
```
Comment (szmoore) [inline contextily/tile.py]: The `os.path.exists` check might not be required in Python 3, but I am used to having to use it from Python 2.

**Comment by darribas** [comment]
Thank you very much for the PRs!!! A bit slow here... @jorisvandenbossche / @ljwolf how does this look to you?

**Comment by jorisvandenbossche** [comment]
The joblib based caching is merged, see https://github.com/darribas/contextily/issues/46#issuecomment-516580038 for some context.

**Comment by uvchik** [comment]
I think this PR can be closed because with PR #85 it is possible to cache tiles using the following line:

```python
import contextily
contextily.tile.memory.store_backend.location = 'my/cache/path' 
```

@szmoore What do you think?

**Comment by darribas** [comment]
Closing as #106 takes this forward.

**Pull request closed without merging**